### PR TITLE
Add codecov to github actions

### PR DESCRIPTION
**Pull request by timcallow**
Trying to add Codecov integration to GitHub actions. 

So far, I followed steps GH0-GH2 [here](https://docs.codecov.com/docs/github-tutorial).

Seems we should first merge this branch into develop, according to the guide linked above. Then we can add more things like minimum coverage checks and a badge.

### CODE REVIEW

**Comment by DanielKotik** [comment]
Do you really need a CodeCav account? Even if you only want to have the CodeCove badge in the README (GH3)?

**Comment by timcallow** [comment]
> Do you really need a CodeCov account? Even if you only want to have the CodeCov badge in the README (GH3)?

Don't see how this would be possible, CodeCov won't give you a badge url without setting everything else up first.

**Review by DanielKotik — APPROVED** [review]
Can you then just add the badge as well?

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 [![PyPI version](https://badge.fury.io/py/atoMEC.svg)](https://badge.fury.io/py/atoMEC)
 [![Python 3.8](https://img.shields.io/badge/python-3.8-blue.svg)](https://www.python.org/downloads/release/python-380/)
+[![codecov](https://codecov.io/gh/atomec-project/atoMEC/branch/develop/graph/badge.svg?token=V66CJJ3KPI)](https://codecov.io/gh/atomec-project/atoMEC)
```
Comment (DanielKotik) [inline README.md:11]: Is the  `token=.....` actually needed? I guess as soon as the PR is merged, the badge dor the `develop` branch will work as it does currently for the `codecov` branch: 

- https://codecov.io/gh/atomec-project/atoMEC/branch/codecov/graph/badge.svg → 
- https://codecov.io/gh/atomec-project/atoMEC/branch/develop/graph/badge.svg →

Comment (DanielKotik) [reply README.md:11]: If they provide the link with this token, its fine. In scikitlearn, a token is also attached, see https://raw.githubusercontent.com/scikit-learn/scikit-learn/main/README.rst . I think it doesn't really matter. OK, you can merge it!

**Pull request merged**